src/autocomplete/WildcardMentionItem.js

```diff
@@ -91,42 +95,42 @@ const descriptionOf = (
 // > https://zulip.com/help/mention-a-user-or-group
 // > https://zulip.com/help/pm-mention-alert-notifications
 // > So I think the right order of preference is [all, everyone, stream].
-const kOrderedTypesWithStream = [
+//
+// ("topic" could probably go equally well before or after those three.)
+const kOrderedTypesWithStreamAndTopic = [
   WildcardMentionType.All,
   WildcardMentionType.Everyone,
   WildcardMentionType.Stream,
+  WildcardMentionType.Topic,
 ];
 
 /**
- * The enum's members, as an array in order of preference, without "stream".
+ * The enum's members, as an array in order of preference, without "stream" and "topic".
  *
- * When a query matches more than one of these, choose the first one. For a
- * similar array that includes WildcardMentionType.Stream, see
- * kOrderedTypesWithStream.
+ * For a similar array that includes WildcardMentionType.Stream and
+ * WildcardMentionType.Topic, see kOrderedTypesWithStreamAndTopic.
  */
 // See implementation note at kOrderedTypesWithStream.
-const kOrderedTypesWithoutStream = [WildcardMentionType.All, WildcardMentionType.Everyone];
+const kOrderedTypesWithoutStreamAndTopic = [WildcardMentionType.All, WildcardMentionType.Everyone];
 
 export const getWildcardMentionsForQuery = (
   query: string,
   destinationNarrow: Narrow,
   _: GetText,
-): $ReadOnlyArray<WildcardMentionType> => {
-  // This assumes that we'll never want to show two suggestions for one query.
-  // That's OK, as long as all of WildcardMentionType's members are synonyms,
-  // and it's nice not to crowd the autocomplete with multiple items that mean
-  // the same thing. But we'll need to adapt if it turns out that all the
-  // members aren't synonyms.
-  const match = (
-    isStreamOrTopicNarrow(destinationNarrow) ? kOrderedTypesWithStream : kOrderedTypesWithoutStream
-  ).find(
+): $ReadOnlyArray<WildcardMentionType> =>
+  // Since not all of WildcardMentionType's members are synonyms -- "topic"
+  // has a different meaning than "all", "everyone", and "stream" -- show
+  // all matches instead of just the first. That's so e.g. a query for
+  // "topic" in the user's language can't be short-circuited by the "all"
+  // result.
```

Comment:
Hmm, this means a little UX regression in that the results will now be crowded with several items that mean the same thing.

I think it won't be too complicated to maintain our existing nice behavior of suppressing dupes, while not inappropriately suppressing `@topic`. The main thing to do in the code is that in retrospect the existing version probably stretches too hard to make it table-driven, with those constants `kOrderedTypes…`, rather than writing out logic in code in the function. If we do the latter, it becomes pretty straightforward to add `@topic` without losing dupe suppression on the others.

I just wrote up a version of that; I'll push it to the PR branch.